src/interpreter.jl

```diff
@@ -259,7 +288,7 @@ end
 
 function CC.IRInterpretationState(interp::CthulhuInterpreter,
     code::CodeInstance, mi::MethodInstance, argtypes::Vector{Any}, world::UInt)
-    inferred = @atomic :monotonic code.inferred
+    inferred = code.inferred
```

Comment:
That was a change from your original PR, is `CodeInstance` guaranteed to be "owned" by the current compilation? Would that be what `engine_reserve`/`engine_reject` do?

Reply:
If you declare a field with `@atomic`, the necessary fence will be automatically added in its `getfield` operation. So, you don't need the `@atomic` annotation at the call site. We can leave it there for clarity, though.

Reply:
Ah yes, I see, it's already taken care of in the struct declaration. No need to think about it there then!